lib/access_token_agent/connector.rb

```diff
@@ -10,20 +12,32 @@ def initialize(host:,
       @client_id = client_id
       @client_secret = client_secret
       @fake_auth = fake_auth
+      set_default_config!
     end
 
-    def authenticate
+    def http_auth_header
+      { Authorization: "Bearer #{token}" }
+    end
+
+    def token
       return if @fake_auth
-      fetch_token unless @known_token && @known_token.valid?
+      fetch_token! unless @known_token && @known_token.valid?
```

Comment:
I'd drop the bang.
If there is no method without the bang that behaves differently (less destructive) I don't see the point in marking some methods with a bang unless you want to mark every method that is not purely functional.